primitives/io/src/lib.rs

```diff
@@ -279,7 +279,33 @@ pub trait DefaultChildStorage {
 		storage_key: &[u8],
 	) {
 		let child_info = ChildInfo::new_default(storage_key);
-		self.kill_child_storage(&child_info);
+		self.kill_child_storage(&child_info, None);
+	}
+
+	/// Clear at most `limit` keys from the specified child storage.
+	///
+	/// Deletes all keys from the overlay and up to `limit` keys from the backend.
+	/// This can be used to partially delete a child trie in case it is too large
+	/// to delete in one go (block). It returns false iff some keys are remaining in
+	/// the child trie after the functions returns.
+	///
+	/// Please note that keys that are residing in the overlay for that child tie when
+	/// issuing this call are all deleted without counting towards the `limit`. Only keys
+	/// written during the current block are part of the overlay.
+	///
+	/// # Note
+	///
+	/// Calling this function multiple times per block for the same `storage_key` does
+	/// not make much sense because it is not cumulative when called inside the same block.
+	/// Use this function to distribute the deletion of a single child trie across multiple
+	/// blocks.
+	fn storage_kill_limited(
```

Comment:
Another possible name could be `storage_kill_firsts`.

Could also consider using a new version of existing 'storage_kill' with optional parameter, I have no real opinion about it (in both case it is a new host function, it would just reduce the api).

Reply:
Yeah, why not merge both calls?

@cheme I always thought that the idea of child tries also was that dropping them is constant? Is this some feature for the future?

Reply:
I was under the impression that changing existing runtime interfaces is a no go: Wouldn't that require updating the runtime and client at exactly the same time?

Reply:
Runtime interface supports versioning, meaning you could just add version 2 of the function. The only requirement in this case would be that nodes are updated before a runtime update is applied.

Reply:
Wasn't aware of that. Then I guess we should do that, instead.

Reply:
> @cheme I always thought that the idea of child tries also was that dropping them is constant? Is this some feature for the future?

That was mainly true with rocksdb where we could just call rocksdb primitive delete by prefix directly on the trie nodes, with paritydb there is no key ordering so it would need a trie iteration.
Still paritydb #5280 was relevant because it allows doing the removal during chain pruning only (and not at all for archive node) which can easily be done in an asynch way.
One could also dedicate a db column for a child trie, but that does not seems very suitable for existing dbs.
This was implemented in #5280 for both db, but the pr end up closed.
One of the issue that I remember from it was that it leeks the child trie abstraction a bit everywhere (which seems needed, next step would have been db support a 'subcollection' or 'dynamic column' for child trie that support a drop).

Reply:
I replaced it by version 2 of `kill_storage`.